challenge_1/rust/makernaren/src/string_reverser.rs

```diff
@@ -0,0 +1,34 @@
+/*Not an efficient solution, but I learnt arrays, vectors in rust and can't 
+define run time array in rust :( */
+pub fn reverse(input: &str) -> String {
+	let n = input.len();
+	// Define a vector of char type with length of the input string
+	let mut char_vec = vec!['a'; n];
+	// Replace the chars to tail of the vector
+	for (i, c) in input.chars().enumerate() {
+		char_vec[n-1-i] = c
+		// char_array.push(c)
+	}
+	// Iterate, copy and collect it as a string
+	let result: String = char_vec.iter().cloned().collect();
+	return result
+	// oneliner : return input.chars().rev().collect::<String>();
```

Comment:
Expression semantics in a language that looks like C took me a *long* time to get used to, honestly.